src/mp_units.cpp

```diff
@@ -0,0 +1,74 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+module;
+
+#include <cassert>
+#include <gsl/gsl-lite.hpp>
+
+export module mp_units;
+
+export import <array>;
+export import <compare>;
+import <optional>;
+import <ratio>;
+import <tuple>;
```

Comment:
Other than `<compare>`, I'm not sure why these are necessary. They come from the currently standalone `magnitude.h`.

Reply:
I am not sure what you mean here?

Reply:
Without these, compiling the module during the processed `magnitude.h` or `prime.h` breaks, and for the exported `<array>`, also on use of the static `std::array` data member in the tests. No other `mp_units` header required me to `import` a C++ standard library header (besides the exported `<compare>` on uses of `<=>` by importers).